pkg/kp/consulutil/watch_test.go

```diff
@@ -448,3 +449,82 @@ func TestWatchNewKeysExistingData(t *testing.T) {
 		t.Error("error picking up existing data")
 	}
 }
+
+func TestWatchDiff(t *testing.T) {
+	t.Parallel()
+	f := consultest.NewFixture(t)
+	defer f.Stop()
+
+	done := make(chan struct{})
+	defer func() {
+		if done != nil {
+			close(done)
+		}
+	}()
+	kv1a := &api.KVPair{Key: "prefix/hello", Value: []byte("world")}
+	kv1b := &api.KVPair{Key: "prefix/hello", Value: []byte("computer")}
+	kv2a := &api.KVPair{Key: "prefix/test", Value: []byte("foo")}
+	kv3a := &api.KVPair{Key: "something", Value: []byte("different")}
+
+	// Process existing data
+	var changes *WatchedChanges
+	watchedCh := make(chan *WatchedChanges)
+
+	f.Client.KV().Put(kv1a, nil)
+	go WatchDiff("prefix/", f.Client.KV(), watchedCh, done, testLogger(t))
+
+	changes = <-watchedCh
```

Comment:
probably select on this with a `<-time.After(1*time.Second)` so that if a value never arrives, the test doesn't hang